float8_experimental/float8_linear_utils.py

```diff
@@ -117,8 +117,23 @@ def swap_linear_with_float8_linear(
             swap_linear_with_float8_linear(child, module, emulate)
 
 
+def get_float8_layers(model: torch.nn.Module, fp8_classes=None):
+    if fp8_classes is None:
+        # Lazy import to avoid circular dependency
+        from float8_experimental.float8_linear import Float8Linear
+
+        fp8_classes = Float8Linear
+
+    # Get all fp8 layers and tensors
+    fp8_layers = [
+        child for name, child in model.named_modules() if isinstance(child, fp8_classes)
+    ]
+
+    return fp8_layers
+
+
 def sync_float8_amax_and_scale_history(
-    model: torch.nn.Module, fp8_classes=None
+    model: torch.nn.Module, fp8_classes=None, fp8_layers=None, combine_reduction=False
```

Comment:
can we measure time on single GPU, and if that's a net positive as well just delete the old code path?  It would be great to keep things simple.

Reply:
Did you mean the `combine_reduction` option or the `fp8_layers` option? 

For `combine_reduction`, I think we can remove it and keep `combine_reduction=True` as default. 

For `fp_layers`, we have many exiting tests and benchmarks (such as single-gpu llama_7B benchmarks) that using the original call (`sync_float8_amax_and_scale_history(model)`). So I kept it as optional and can support `None`